packages/toolpad-app/src/toolpad/AppState.tsx

```diff
@@ -980,6 +980,7 @@ export default function AppProvider({ appUrl, children }: DomContextProps) {
         dispatch({ type: 'DOM_SAVED', savedDom: domToSave });
       })
       .catch((err) => {
+        console.error(err.message);
```

Comment:
```suggestion
        console.error(`Failed to save the dom: ${err.message}`);
```

Reply:
How do you feel about the following
- when we detect that prettier errored, we show confirmation a dialog "prettier failed with error X, do you want to save without formatting".
- this sets a `ignoreFormattingErrors` to the `state`
- add a `force: boolean` option to `applyDomDiff` which tries to prettify but ignores errors, we call it with:
    ```tsx
    applyDomDiff(state.dom, { force: state.ignoreFormattingErrors })
    ```

Reply:
let me try

Reply:
> How do you feel about the following
> 
> * when we detect that prettier errored, we show confirmation a dialog "prettier failed with error X, do you want to save without formatting".
> * this sets a `ignoreFormattingErrors` to the `state`
> * add a `force: boolean` option to `applyDomDiff` which tries to prettify but ignores errors, we call it with:
>   ```tsx
>   applyDomDiff(state.dom, { force: state.ignoreFormattingErrors })
>   ```

>when we detect that prettier errored, we show confirmation a dialog "prettier failed with error X, do you want to save without formatting".

Do you have a suggestion for a dialog that I can directly use in the project

Reply:
>when we detect that prettier errored, we show confirmation a dialog "prettier failed with error X, do you want to save without formatting".

how to display this dialog ? if a format error is detected, this `dialog` will be always displayed 
 is there a strategy for this dialog?

Reply:
@Janpot

Reply:
> if a format error is detected, this dialog will be always displayed
> is there a strategy for this dialog?

If a user confirms the dialog, then `ignoreFormattingErrors` will get set in the state, and the next time `applyDomDiff` is called it will get a `force: true` value and shouldn't error on formatting errors.

Reply:
We don't leave that option open. You either confirm or fix your setup.